rpc_test.go

```diff
@@ -85,7 +86,36 @@ func TestPingPong(t *testing.T) {
 	assert.NotEmpty(t, r2.ID)
 	assert.Equal(t, p1.Host().ID().String(), r2.From.String())
 
+	// test retries; peer1 requests "pong" from peer2, but peer2 joins topic after the request
```

Comment:
Is it the retry logic for? It feels to me that with relatively long `retryWait` and/or > 1 backoff, there's still a high chance that a new peer joins before deadline is hit and doesn't get the message. An alternative would be to re-deliver all ongoing messages whenever a new peer subscribing the topic becomes online, which would be more responsive. That feels to be the responsibility of `go-libp2p-pubsub`, but also doable by this package.

Reply:
Yeah, I think you have me convinced that retries aren't the right approach for pubsub / mesh network based things that have an out-of-band "room". I don't have a deep understanding the protocol, but subsequent retries aren't likely to _be the cause of another peer joining the room_, which makes them worthless.